lib/spack/spack/util/executable.py

```diff
@@ -21,11 +20,14 @@ class Executable:
     """Class representing a program that can be run on the command line."""
 
     def __init__(self, name):
-        # necesary here for the shlex call to succeed
-        name = format_os_path(name, mode=Path.unix)
-        self.exe = shlex.split(str(name))
-        # filter back to platform dependent path
-        self.exe = path_to_os_path(*self.exe)
+        file_path = which_string(name)
```

Comment:
The `shlex.split` was there for if someone does `Exectuable("ls /an/arg/path")`: we will still want to handle that here (apparently our tests are lacking for it though)

Reply:
Notes: 

Change package executable calls where spaces to arguments are passed in the constructor. Split them by arguments now instead. Check to see if when the subprocess fails, and there's a space, give error asking if they meant to space it the way they did.

Remove which from init function. Too slow.

Reply:
@scheibelp All of the recommend changes are applied. Let me know what you think!